pennylane/vqe/vqe.py

```diff
@@ -175,6 +252,48 @@ def wires(self):
     def name(self):
         return "Hamiltonian"
 
+    @property
+    def grouping_indices(self):
+        """Return the grouping indices attribtue.
+
+        Returns:
+            list[list[int]]: indices needed to form groups of commuting observables
+        """
+        return self._grouping_indices
+
+    def compute_grouping(self, grouping_type="qwc", method="rlf"):
+        """
+        Compute groups of indices corresponding to commuting observables of this Hamiltonian, and store it in the `grouping_indices` attribute.
+
+        Args:
+            grouping_type (str): The type of binary relation between Pauli words used to compute the grouping.
+                Can be ``'qwc'``, ``'commuting'``, or ``'anticommuting'``. . Ignored if compute_grouping is False.
+            method (str): The graph coloring heuristic to use in solving minimum clique cover for grouping, which
+                can be ``'lf'`` (Largest First) or ``'rlf'`` (Recursive Largest First). Ignored if compute_grouping is False.
+        """
+
+        self._grouping_indices = qml.transforms.invisible(_compute_grouping_indices)(
+            self.ops, grouping_type=grouping_type, method=method
+        )
+
+    def get_grouping(self):
+        """Return groupings of commuting observables and their corresponding coefficients.
+
+        Returns:
+            list[tensor_like], list[list[Observable]]: groups of coefficients and observables
+        """
+        if self._grouping_indices is None:
+            # if it has not been done before, compute which indices of coefficients/observables belong to each group
+            self._grouping_indices = qml.transforms.invisible(_compute_grouping_indices)(self.ops)
+
+        grouped_coefficients = [
+            qml.math.squeeze(qml.math.take(self.coeffs, indices, axis=0))
+            for indices in self._grouping_indices
+        ]
+        grouped_observables = [[self.ops[i] for i in indices] for indices in self._grouping_indices]
+
+        return grouped_coefficients, grouped_observables
```

Comment:
I could delete the method above and rather explicitely do the extraction in `hamiltonian_expand`? Or is it nice to have here? It's nothing a user could not do herself or get from the grouping module!